### DISCUSSION TITLE
How to insert NOW() or current_timestamp?

### DISCUSSION THREAD

**Discussion by shanicky** [SeaQL/sea-orm#1859] [Q&A]
Hello, 

We’ve recently encountered a problem. We use the `NOW()` function for some detailed recording operations, but I don’t know how to set an expr `NOW()` in ActiveModel for insert operations. 

Could you tell me what to do? Thank you.

**Comment by tyt2y3** [comment]
Try https://docs.rs/sea-query/latest/sea_query/expr/struct.Expr.html#method.current_timestamp ?

**Reply by shanicky** [reply]
Thank you for your response, but what I am aiming for is to implement the usage similar to `insert into table (col) values (now()) ` through Active Model.

**Reply by dave42w** [reply]
I have added this to the end of my entity sources Is this what you need?

```
use chrono::Utc;
use sea_orm::{prelude::async_trait::async_trait, ActiveValue};

#[async_trait]
impl ActiveModelBehavior for ActiveModel {
    async fn before_save<C: ConnectionTrait>(
        mut self,
        _db: &C,
        insert: bool,
    ) -> Result<Self, DbErr> {
        if insert {
            self.created_timestamp = ActiveValue::Set(Utc::now().to_string());
        };
        self.updated_timestamp = ActiveValue::Set(Utc::now().to_string());
        Ok(self)
    }
}
```
